cmd/root.go

```diff
@@ -22,42 +23,97 @@ var config configpkg.ConfigType
+		viper.BindPFlag("graceful-success-time", cmd.Flags().Lookup("graceful-success-time"))
+		viper.BindPFlag("downloader-entrypoint", cmd.Flags().Lookup("downloader-entrypoint"))
+		viper.BindPFlag("executor-entrypoint", cmd.Flags().Lookup("executor-entrypoint"))
+		viper.BindPFlag("uploader-entrypoint", cmd.Flags().Lookup("uploader-entrypoint"))
+		viper.BindPFlag("opi-entrypoint", cmd.Flags().Lookup("opi-entrypoint"))
+
+	},
 	Run: func(cmd *cobra.Command, args []string) {
 		var err error
+		webhookHost := viper.GetString("operator-webhook-host")
+		webhookPort := viper.GetInt32("operator-webhook-port")
+		serviceName := viper.GetString("operator-service-name")
+		webhookNamespace := viper.GetString("operator-webhook-namespace")
+		register := viper.GetBool("register")
+		gracefulFailTime := viper.GetString("graceful-fail-time")
+		gracefulSuccessTime := viper.GetString("graceful-success-time")
+		downloaderEntrypoint := viper.GetString("downloader-entrypoint")
+		executorEntrypoint := viper.GetString("executor-entrypoint")
+		uploaderEntrypoint := viper.GetString("uploader-entrypoint")
+		opiEntrypoint := viper.GetString("opi-entrypoint")
 
 		LogDebug("Namespace: ", config.Namespace)
 		LogDebug("Loggregator-endpoint: ", config.LoggregatorEndpoint)
 		LogDebug("Loggregator-ca-path: ", config.LoggregatorCAPath)
 		LogDebug("Loggregator-cert-path: ", config.LoggregatorCertPath)
 		LogDebug("Loggregator-key-path: ", config.LoggregatorKeyPath)
+
+		LogDebug("Webhook listening on: ", webhookHost, webhookPort)
+		LogDebug("Webhook namespace: ", webhookNamespace)
+		LogDebug("Webhook serviceName: ", serviceName)
+		LogDebug("Webhook register: ", register)
+
 		LogDebug("Starting Loggregator")
+		if webhookHost == "" {
+			LogDebug("required flag 'operator-webhook-host' not set (env variable: OPERATOR_WEBHOOK_HOST)")
+		}
 
+		RegisterWebhooks := true
+		if !register {
+			LogDebug("The extension will start without registering")
+			RegisterWebhooks = false
+		}
 		err = config.Validate()
 		if err != nil {
 			LogError(err.Error())
 			os.Exit(1)
 		}
 
 		filter := false
-
+		ctx := context.Background()
 		x := eirinix.NewManager(eirinix.ManagerOptions{
 			Namespace:           config.Namespace,
 			KubeConfig:          kubeconfig,
+			Context:             &ctx,
 			OperatorFingerprint: "eirini-loggregator-bridge", // Not really used for now, but setting it up for future
 			FilterEiriniApps:    &filter,
+
+			Host:             webhookHost,
+			Port:             webhookPort,
+			ServiceName:      serviceName,
+			WebhookNamespace: webhookNamespace,
+			RegisterWebHook:  &RegisterWebhooks,
 		})
 
 		pw := podwatcher.NewPodWatcher(config)
 		// Setup does need the manager to get kubernetes connection
-		if err := pw.EnsureLogStream(x); err != nil {
+		if err := pw.EnsureLogStream(ctx, x); err != nil {
 			LogError(err.Error())
 			os.Exit(1)
 		}
 
-		x.AddWatcher(pw)
+		x.AddExtension(podwatcher.NewGracePeriodInjector(&podwatcher.GraceOptions{
```

Comment:
well yes, it's a new thing available on EiriniX, but at the same time it would fail immediately anyhow.  Addressed in https://github.com/cloudfoundry-incubator/eirini-loggregator-bridge/pull/26/commits/7887894217194e764b8634c7b6857d4e9b915a9b